core/build.gradle

```diff
@@ -137,6 +137,12 @@ dependencies {
 
     // Jetty, for testing interoperability with other servers.
     testImplementation 'org.eclipse.jetty:jetty-webapp'
+
+    // Brotli
+    implementation "com.aayushatharva.brotli4j:brotli4j:1.4.2"
+    testImplementation "com.aayushatharva.brotli4j:native-linux-x86_64:1.4.2"
+    testImplementation "com.aayushatharva.brotli4j:native-osx-x86_64:1.4.2"
+    testImplementation "com.aayushatharva.brotli4j:native-windows-x86_64:1.4.2"
```

Comment:
Shouldn't we pull these in as `implementation` dependency?

Reply:
The native jars seem pretty massive, it would be good if they aren't required for gzip users. Perhaps need an SPI keyed on encoding-type?

Reply:
Ah but the code is guarded by `Brotli.isAvailable` it seems - so even without SPI as long as it's opt-in by adding these dependencies that works well too

Reply:
Though SPI could still be good to not have even the `brotli4j` dependency on core. Armeria core POM is still a good example of only depending on very prominent / core modules. Maybe dangerous to lead to set a precedent of dependency sprawl

https://repo1.maven.org/maven2/com/linecorp/armeria/armeria/1.9.2/armeria-1.9.2.pom

Reply:
I see. Then lets guard this with `Brotli.isAvailable()` and make them optional `implementation` dependencies. What do you think?

Reply:
Should we then use `optionalImplementation` instead of `testImplementation`?